ucan/lib.go

```diff
@@ -5,6 +5,7 @@ import (
 	"time"
 
 	"github.com/ipld/go-ipld-prime/datamodel"
+	"github.com/storacha-network/go-ucanto/core/ipld"
```

Comment:
I wonder if we should just redefine the builder interface here.

Reason:
1. This links the ucan package back into ucanto core, which makes it less easily extractable.
2. There's minimal consequence to doing so -- two interfaces with the same methods are essentially interchangable.

Reply:
re: extractability -- i am in active conversations with folks who want this extracted to become the default go-ucan, especially once it's 1.0

Reply:
> I wonder if we should just redefine the builder interface here.

I did this 👍